src/services/NetworkService.ts

```diff
@@ -168,4 +169,18 @@ export class NetworkService {
                   websocketInjected: WebSocket,
               });
     }
+    /**
+     * It checks if a node has Websocket functioning properly to subscribe.
+     * @param url the url.
+     */
+    public async checkWebsocketConnection(url: string) {
+        const webSocket = new WebSocket(url);
+        let websocketConnectionStatus: boolean = false;
+        webSocket.onmessage = function (e) {
+            // @ts-ignore
+            websocketConnectionStatus = e.currentTarget.readyState == 1;
+        };
+        await CommonHelpers.sleep(2000);
```

Comment:
Why do we need this delay?

Reply:
to make sure readyState has the correct value returned from the function.

Reply:
Can we replace `2000` with a timeout parameter here? (which is defaulted to `2000`)